ui/webui/resources/tools/build_webui.gni

```diff
@@ -0,0 +1,156 @@
+
+  this_path_ = rebase_path(".", "//")
+  print("Brave overriding webui target: $this_path_/$target_name")
+
+  # Generate the list for inclusion via preprocess and exclusion for chromium's preprocess
+  brave_ts_files = []
+  _has_ts = defined(invoker.brave_css_files)
+
+  if (defined(invoker.brave_non_web_component_files)) {
+    _has_ts = true
+
+    # consider the regular (non web component) ts files
+    brave_ts_files += invoker.brave_non_web_component_files
+  }
+
+  brave_html_files = []
+  if (defined(invoker.brave_web_component_files)) {
+    _has_ts = true
+
+    # consider the web component ts files
+    brave_ts_files += invoker.brave_web_component_files
+
+    # consider the .ts version of .html files
+    foreach(component_file, invoker.brave_web_component_files) {
+      brave_html_files += [ string_replace(component_file, ".ts", ".html") ]
+    }
+  }
+
+  # Move brave files to the place where chromium webui TS is compiled from.
+  # It should be the same directory that build_webui.gni uses after it performs
+  # its own preprocess_if_expr. Once chromium and brave frontend files are all in
+  # the same place, build_webui.gni will continue its typescript and rollup
+  # compile (etc.) from that directory.
+  if (_has_ts) {
+    preprocess_if_expr("brave_preprocess") {
+      if (defined(invoker.brave_preprocessor_defines)) {
+        defines = invoker.brave_preprocessor_defines
+      }
+      out_folder = "${target_gen_dir}/preprocessed"
+
+      in_files = brave_ts_files + brave_html_files
+
+      if (defined(invoker.brave_css_files)) {
+        in_files += invoker.brave_css_files
+      }
+    }
+  }
+
+  if (defined(invoker.brave_static_files)) {
+    extras_grdp = "$target_gen_dir/brave_extras.grdp"
+    generate_grd("brave_static_files") {
+      grd_prefix = "brave_extras"
+      out_grd = extras_grdp
+      input_files = invoker.brave_static_files
+      input_files_base_dir = invoker.brave_base_dir
+    }
+  }
+
+  build_webui(target_name) {
+    forward_variables_from(invoker,
+                           "*",
+                           [
+                             "brave_base_dir",
+                             "brave_static_files",
+                             "brave_web_component_files",
+                             "brave_non_web_component_files",
+                             "brave_css_files",
+                             "brave_preprocessor_defines",
+                             "brave_ts_definitions",
+                             "brave_mojo_files",
+                             "brave_mojo_files_deps",
+                             "brave_ts_deps",
+                           ])
+
+    if (defined(invoker.brave_static_files)) {
+      extra_grdp_deps = [ ":brave_static_files" ]
+      extra_grdp_files = [ extras_grdp ]
+    }
+
+    if (defined(invoker.brave_web_component_files)) {
+      web_component_files += invoker.brave_web_component_files
```

Comment:
are these variables always declared? Shouldn't this be:
```gn
if (!defined(web_component_files)) {
  web_component_files  = []
}
web_component_files += invoker.brave_web_component_files
```
?

Reply:
I think `web_component_files` probably isn't defined in the new Lit uis (as its polymer only)